app/liquid_tags/tweet_tag.rb

```diff
@@ -1,7 +1,9 @@
 class TweetTag < LiquidTagBase
   include ActionView::Helpers::AssetTagHelper
   PARTIAL = "liquids/tweet".freeze
-  ID_REGEXP = /\A\d{10,20}\z/ # id must be all numbers between 10 and 20 chars
+  REGISTRY_REGEXP = %r{https://twitter.com/\w{1,15}/status/(?<id>\d{10,20})}
+  VALID_ID_REGEXP = /\A(?<id>\d{10,20})\Z/
+  REGEXP_OPTIONS = [REGISTRY_REGEXP, VALID_ID_REGEXP].freeze
```

Comment:
I'm not super familiar with the internals of this class - is REGEXP_OPTIONS used in a way that _any_ 10-20 digit string is construed as a valid twitter id - and do we have any other (youtube?) embeds with overlapping recognizers?

if I have an 11 digit "tweet" id, and an 11 digit youtube video id, 
  VALID_ID_REGEXP = /\A(?<video_id>[a-zA-Z0-9_-]{11})

which one, if any, does `{% embed 12345678901 %}` choose?

I'll _test_ that in your branch but it's the first thing I saw reading this code.

Reply:
good question, I will look into it

Reply:
answer: no concern here, tweet and youtube tags may take ids but embed does not

Reply:
it looks like I need an api key to fetch a tweet in the api? 

I don't see that in .env so I assume I need to set that up in config?

Reply:
ahh yes @djuber I needed to grab keys from developer.twitter.com and plug them into `config`

Reply:
So, @djuber, you brought up a very good point about the `embed` keyword returning an unexpected embed if the `id` say matches another tag.

For liquid tags that receive just a number(`id`), the keyword is the type of embed expected (e.g. `{% twitter 12345 %}`. So I don't think a mismatched embed will result.

For liquid tags that use the `embed` keyword, the expected input is a URL, which specifies the type of embed to be returned. By the time the `id` is parsed out, the correct embed type has already been chosen from the Registry. So again I don't think a mismatched embed will result.

Let me know if my understanding has any holes. 😅